index.js

```diff
@@ -16,159 +16,221 @@ const isDeflate = require('is-deflate')
 const encodingNegotiator = require('encoding-negotiator')
 
 function compressPlugin (fastify, opts, next) {
-  fastify.decorateReply('compress', compress)
+  const globalParams = processParams(opts)
 
-  if (opts.global !== false) {
-    fastify.addHook('onSend', onSend)
+  if (opts.encodings && opts.encodings.length < 1) {
+    next(new Error('The `encodings` option array must have at least 1 item.'))
+  }
+
+  if (globalParams.encodings.length < 1) {
+    next(new Error('None of the passed `encodings` were supported — compression not possible.'))
+  }
+
+  // add onSend hook onto each route as needed
+  fastify.addHook('onRoute', (routeOptions) => {
+    if (routeOptions.config && typeof routeOptions.config.compress !== 'undefined') {
+      if (typeof routeOptions.config.compress === 'object') {
+        const mergedCompressParams = Object.assign({}, globalParams, processParams(routeOptions.config.compress))
+        // if the current endpoint has a custom compress configuration ...
+        buildRouteCompress(fastify, mergedCompressParams, routeOptions)
+      } else if (routeOptions.config.compress === false) {
+        // don't apply any compress settings
```

Comment:
this case is not documented

Reply:
Added `Note` in the README about being able to exclude compression on a route with this.